src/chain/chain.rs

```diff
@@ -218,7 +217,7 @@ impl Chain {
             Err(InsertError::ReplacedAlreadyInserted) => {
                 // TODO: If detecting duplicate vote from peer, penalise.
                 log_or_panic!(
```

Comment:
I am not sure log_or_panic is valid anywhere except POC work, but less so here. As this is an example we can log errors or panic, but I don't think to mix them is good. In the cases where it seems to log is fine, unless we get into an unusable/UB state.

Reply:
We still use `log_or_panic` in a bunch of places and this PR isn't really about that. I'm OK with replacing `log_or_panic` with something better, but it should be done as a separate task. We also haven't fully decided what that better thing should be.

Reply:
Yes, I agree, we will get to it soon enough. Thanks chap